### DISCUSSION TITLE
What pre-processing should I do on my colored raster image before passing it to vtracer to get best results?

### DISCUSSION THREAD

**Discussion by abhimanyu891998** [visioncortex/vtracer#83] [Q&A]
Hi, was wondering what steps can I take on my raster (png) image before I give it to vtracer for converting to svg. Some that I already have in mind are:

1.) Color quantization of the png - Does it help to reduce colors of the png for svg conversion?
2.) Upscaling of the png - right now, I have png in 720 x 720 resolution, how much should I upscale it to?

Any other recommendations? Would help a lot, thanks!

**Comment by tyt2y3** [comment]
Basically you don't have to do 1. But if the image is noisy, then run it with some denoiser to remove compression artifact would make it much better. Which is something I was thinking about.

**Reply by abhimanyu891998** [reply]
Thanks for the reply! Our use case aligns with the need to be able to color specific parts of the generated svg, so that's why we thought drastically reducing colors in the image would help with vectorization. But it seems like, even with reduced colors, the final vectorized image from vtracer adds certain colors to the palette. Is that the case? Does vtracer add new colors to the image during the vectorization process?

**Reply by tyt2y3** [reply]
no, it doesn't if you have an already quantized input. set color_precision = 8, then it will only build clusters with specific colors. (try the tank image on https://www.visioncortex.org/vtracer/)

otherwise, in the clustering stage, similar colors will be merged together, and the colors will be averaged.
